hw/drivers/nimble/nrf52/src/ble_phy.c

```diff
@@ -773,14 +773,16 @@ ble_phy_get_cur_rx_phy_mode(void)
                                     (phy == BLE_PHY_MODE_CODED_500KBPS)) {
         /*
          * XXX CI field value is bits [2:1] in NRF_RADIO->PDUSTAT which is
-         *     available in SDK v14.0 - it's NRF_RADIO->RESERVED7[0] in
-         *     SDK v11.0 so let's use it for now
+         *     available in nrfx for nrf52840 only - it's NRF_RADIO->RESERVED7[0]
+         *     for other nrf52 vairants.
          */
-        phy = NRF_RADIO->RESERVED7[0] & 0x06 ?
-                                        BLE_PHY_MODE_CODED_500KBPS :
-                                        BLE_PHY_MODE_CODED_125KBPS;
+#if MYNEWT_VAL(BSP_NRF52840)
```

Comment:
This does not seem right. You can have other BSPs with this particular nordic MCU in them, and have this code be built. Are the original #ifdef's not sufficient?